tests/amor/utils_test.py

```diff
@@ -0,0 +1,248 @@
+# SPDX-License-Identifier: BSD-3-Clause
+# Copyright (c) 2024 Scipp contributors (https://github.com/scipp)
+# flake8: noqa: F403, F405
+
+import pytest
+import scipp as sc
+
+from ess.amor.utils import q_theta_figure, wavelength_theta_figure, wavelength_z_figure
+
+
+@pytest.fixture
+def da():
+    return sc.DataArray(
+        data=sc.ones(dims=('events',), shape=(10,)),
+        coords={
+            'wavelength': sc.linspace('events', 1, 12, 10, unit='angstrom'),
+            'theta': sc.linspace('events', 0, 1, 10, unit='deg'),
+            'Q': sc.linspace('events', 0, 0.3, 10, unit='1/angstrom'),
+            'z_index': sc.arange('events', 0, 10),
+        },
+    )
+
+
+@pytest.fixture
+def wavelength_bins():
+    return sc.linspace('wavelength', 1, 12, 3, unit='angstrom')
+
+
+@pytest.fixture
+def theta_bins():
+    return sc.linspace('theta', 0, 1, 3, unit='deg')
+
+
+@pytest.fixture
+def q_bins():
+    return sc.linspace('Q', 0, 0.3, 3, unit='1/angstrom')
+
+
+def test_wavelength_figure_table(da, wavelength_bins, theta_bins):
+    with pytest.raises(ValueError, match='binning provided'):
+        wavelength_theta_figure(da)
+
+    with pytest.raises(ValueError, match='binning provided'):
+        wavelength_theta_figure(da, wavelength_bins=wavelength_bins)
+
+    with pytest.raises(ValueError, match='binning provided'):
+        wavelength_theta_figure(da, theta_bins=theta_bins)
+
+    assert wavelength_theta_figure(
+        da, theta_bins=theta_bins, wavelength_bins=wavelength_bins
+    )
+
+
+def test_wavelength_figure_binned(da, wavelength_bins, theta_bins):
+    with pytest.raises(ValueError, match='binning provided'):
+        wavelength_theta_figure(da.bin(wavelength=3))
+
+    with pytest.raises(ValueError, match='binning provided'):
+        wavelength_theta_figure(da.bin(theta=3))
+
+    assert wavelength_theta_figure(da.bin(wavelength=3, theta=3))
+    assert wavelength_theta_figure(da.bin(wavelength=3), theta_bins=theta_bins)
+    assert wavelength_theta_figure(da.bin(theta=3), wavelength_bins=wavelength_bins)
+
+
+def test_wavelength_figure_hist(da, wavelength_bins, theta_bins):
+    with pytest.raises(ValueError, match='must have wavelength and theta coord'):
+        wavelength_theta_figure(da.hist(wavelength=3))
+
+    with pytest.raises(ValueError, match='must have wavelength and theta coord'):
+        wavelength_theta_figure(da.hist(theta=3))
+
+    assert wavelength_theta_figure(da.hist(wavelength=3, theta=3))
```

Comment:
Not sure (here and in other tests) you need the `assert`s?

Reply:
No you're right they are not really needed. But they are used to communicate that this is where the test happens, this line is not part of the setup.

If you think that makes it confusing I'll remove the asserts.

Reply:
I don't really mind.